src/lib/Weasyprint.ts

```diff
@@ -0,0 +1,49 @@
+import { Config, WeasyprintConfig } from '../types';
+import { Request, Response } from 'express';
+import { spawn } from 'child_process';
+import { PrintInput } from '../types';
+
+export class WeasyprintParams {
+  #config: object;
+
+  constructor(config ?: WeasyprintConfig) {
+    this.#config = config || {};
+  }
+
+  toArgs(): string[] {
+    return Object
+      .entries(this.#config)
+      .reduce((memo: string[], pair) => memo.concat([`-${pair[0]}`, pair[1]]), [])
+      .concat(['-', '-']);
+  }
+}
+
+export function Weasyprint(config: Config, req: Request, res: Response) {
```

Comment:
Oh? Was the npm wrapper for weasyprint not working properly? I'm surprised you reimplemented a wrapper around the cli.

Potential improvement: I'd take a WritableStream (https://nodejs.org/api/stream.html#class-streamwritable) as as parameter, it would make it more generic. (Response extends WritableStream)